packages/core-db/test/db/sequential-processing-data-service.dbspec.ts

```diff
@@ -0,0 +1,240 @@
+import '../setup'
+
+import {
+  SequentialProcessingDataService,
+  SequentialProcessingItem,
+} from '../../src/types/queue'
+import { Row } from '../../src/types/db'
+import {
+  DefaultSequentialProcessingDataService,
+  PostgresDB,
+} from '../../src/app'
+
+const sequenceKey = 'test'
+const testItem = 'test item'
+const tableName = 'sequential_processing'
+
+describe('DB Test: SequentialProcessingDataService (Requires Postgres!)', () => {
+  let postgres: PostgresDB
+  let dataService: SequentialProcessingDataService
+  before(() => {
+    postgres = new PostgresDB('0.0.0.0', 5432, 'test', 'test', 'rollup')
+    dataService = new DefaultSequentialProcessingDataService(postgres)
+  })
+
+  beforeEach(async () => {
```

Comment:
Could the `beforeEach` and `after` be combined into a single `afterEach`?

Reply:
Unfortunately I don’t think so. The `beforeEach()` is necessary to make sure the test starts with a clean slate. The `after()` is more of a courtesy, getting rid of test state in case the separately running DB will be used for something else after.

It all stems from the fact that the test suite itself does not spin up / down a Postgres instance, so I’m trying to make 0 assumptions about the state before the tests are run.